### PR TITLE
[BB-5168] Add support for JWK Keysets

### PR DESCRIPTION
**Pull request by felipetrz**
This adds support for specifying a provider's keyset URL in the LTI consumer XBlock.

### Testing

1. Have a functioning koa or lilac devstack
2. Setup a LTI 1.3 provider that's able to access the devstack (i.e. via tunneling or provider self-hosting)
2. Install the LTI Consumer XBlock as per the instructions in the Readme, checking out this branch
3. Via the Studio, add an Unit containing an LTI Consumer block.
4. Open the XBlock edit dialog, and change the LTI Version to 1.3.
5. Verify that there's a new option to choose between using a Public Key and a Keyset URL.
6. Select Keyset URL, and fill the Launch URL, Login URL, Access Token URL and Keyset URL to the ones from the provider chosen at step 2.
7. Register the endpoints given by the LTI Consumer XBlock at the Studio preview with the provider.
7. Publish the Unit, and verify it launches successfully from the learning app.

### CODE REVIEW

File: lti_consumer/lti_1p3/key_handlers.py
Comment (giovannicimolin) [inline lti_consumer/lti_1p3/key_handlers.py:73]: @felipetrz This will fail quality checks (exception too generic). Can you add and exception here?

File: lti_consumer/tests/unit/test_lti_xblock.py
Comment (giovannicimolin) [inline lti_consumer/tests/unit/test_lti_xblock.py]: @felipetrz You are reusing a lot of variations for this section, can you push them to a separate method? Or at least the parts of it that are re-usable. You can set up patchers directly on the `setUp` method if that makes testing easier.

**Review by giovannicimolin — CHANGES_REQUESTED** [review]
@felipetrz Minor nits. I'll test the PR once I finish reprovisioning my devstack.

File: lti_consumer/lti_1p3/key_handlers.py
Comment (felipetrz) [reply lti_consumer/lti_1p3/key_handlers.py:73]: @giovannicimolin This does not fail pylint checks because I'm raising another chained exception from the general Exception, in this case.
Adding a disable here results in a `useless-suppression` warning.

File: lti_consumer/tests/unit/test_lti_xblock.py
Comment (felipetrz) [reply lti_consumer/tests/unit/test_lti_xblock.py]: @giovannicimolin I refactored and cleaned the tests up.

**Review by giovannicimolin — APPROVED** [review]
:+1:

- [x] I tested this:
1. Tested a LTI launch using the RSA key and it worked :heavy_check_mark: 
2. Tested a LTI launch using a JWK Keyset url and it worked perfectly as well :heavy_check_mark: 
- [x] I read through the code: 1 minor nit :warning: 
- [ ] ~~I checked for accessibility issues~~ NA
- [ ] ~~Includes documentation~~ NA

@felipetrz Great work here! This is working perfectly! :rocket: 

I'm conditionally approving this ticket, once you resolve the item below you can merge & deploy to UQ's branch.

- The translations files were not updated with the latest strings, run `make extract_translations`, `make build_dummy_translations` and `make compile_translations` and commit the results.

Make sure you use the "Squash and merge" feature and leave everything on a single commit. Can you do the same with the **upstream PR** as well?

**Pull request merged**
